daemon/appinfo.c

```diff
@@ -115,6 +121,7 @@ gchar       *appinfo_to_string (const appinfo_t *self);
 
 bool            appinfo_valid       (const appinfo_t *self);
 control_t      *appinfo_control     (const appinfo_t *self);
+const config_t *appinfo_config      (const appinfo_t *self);
```

Comment:
Two forward declarations of the same function look strange at least (already declared in `daemon/appinfo.h`).
The same about `applications_config()`.

Reply:
Yes, you can say that. What would you like me to do about it?

C has headers which nowadays are basically a workaround for missing proper module framework that more modern languages have. In sailjaild (just like in some other components in Sailfish OS) we have decided to add forward declarations to both source files and their corresponding header files. The difference is that header doesn't declare static functions as it only defines the public interface. This way you can see from source file which functions it implements without checking both files. It's a little more to keep up to date while implementing things but it ought to be more convenient when reading the sources.

Reply:
So there is no any reason in such way.
Ok, I can just assume that it's just a style of this particular project if you insist. Whether it's convenient or not - just a matter of tastes.